src/modules/sway/workspaces.cpp

```diff
@@ -107,11 +107,16 @@ void Workspaces::onCmd(const struct Ipc::ipc_response &res) {
         auto payload = parser_.parse(res.payload);
         workspaces_.clear();
         std::vector<Json::Value> outputs;
+        bool alloutputs = config_["all-outputs"].asBool();
         std::copy_if(payload["nodes"].begin(), payload["nodes"].end(), std::back_inserter(outputs),
-                     [&](const auto &workspace) {
-                       return !config_["all-outputs"].asBool()
-                                  ? workspace["name"].asString() == bar_.output->name
-                                  : true;
+                     [&](const auto &output) {
+                       if (alloutputs && output["name"].asString() != "__i3") {
+                         return true;
+                       }
+                       if (output["name"].asString() == bar_.output->name) {
```

Comment:
Mhmm we should check !alloutputs here otherwise the logic will be the same as before.

Reply:
the only way for it to reach here when `!allouput` is  `true` is when `output["name"]` is `__i3` which will not match `bar_.output->name`